pytorch_lightning/core/lightning.py

```diff
@@ -1372,6 +1372,23 @@ def training_step(...):
         self.trainer.fit_loop.training_loop.batch_loop.backward(loss, optimizer=None, opt_idx=None, *args, **kwargs)
         self._running_manual_backward = False
 
+    def on_before_backward(self, loss: torch.Tensor) -> None:
```

Comment:
Move this to pytorch_lightning/core/hooks.py

Reply:
Done !